captain/utils/test_sequencer/generator/generate_test.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+from captain.models.test_sequencer import StatusTypes, Test, TestTypes
+from captain.parser.bool_parser.utils.name_validator import validate_name
+from captain.utils.logger import logger
+import os
+from openai import OpenAI
+from captain.utils.test_sequencer.data_stream import _with_error_report
+
+key = "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"  # TODO have this from .env, Joey said to hard code for now until his PR
+client = OpenAI(api_key=key)
+
+
+@_with_error_report
+async def generate_test(
+    test_name: str, test_type: TestTypes, prompt: str, test_container: dict
+):
+    if test_type.value != "Python":  # for now only handle python tests
+        raise Exception("Only Python tests allowed")
+    validate_name(test_name)  # this raises an error if invalid
+    response = client.chat.completions.create(
+        model="gpt-3.5-turbo",
+        messages=[
+            {
+                "role": "system",
+                "content": "You generate only raw python code for tests using assertions. If the user's request does not imply a test, simply output 'NULL'",
+            },
+            {
+                "role": "user",
+                "content": prompt,
+            },
+        ],
+    )
+    code_resp = response.choices[0].message.content
+    if code_resp is None:
+        raise Exception("Unable to generate code")
+    if code_resp == "NULL":
+        raise Exception("Invalid prompt/request, please specify something to test")
```

Comment:
Should we check if it's valid Python code ?

Reply:
I have tried it many many times and not a single time has it not generated python code. However, it is stats after all and there is probably a very small probability it messes up. Might be worth it, need a second opinion on this

Reply:
Yeah, it might be a total waste of time to do so, the user can consult the code and edit it manually if needed